modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ManagementToolbar.js

```diff
@@ -82,6 +83,10 @@ function ManagementToolbar({
 	);
 	const [active, setActive] = useState(initialCheckboxStatus !== 'unchecked');
 	const [searchMobile, setSearchMobile] = useState(false);
+	const normalizedViewTypeItems = useMemo(
+		() => normalizeDropdownItems(viewTypeItems),
+		[viewTypeItems]
+	);
```

Comment:
I would normally use a `ref` here, as a way to have a value that is calculated once and then stable in all subsequent renders. Something like:
```js
const normalizedViewTypeItems = useRef(
	normalizeDropdownItems(viewTypeItems)
);
```
and then `normalizedViewTypeItems.current`s below.

Is there a reason why using `useMemo` would be a better solution?

Reply:
Hey @markocikos 

If we use a ref like in your example the normalizeDropdownItems would be calle each render :( this is why I used the useMemo

If we want to do it with a ref we'd need something like this:

```javascript
const normalizedViewTypeItems = useRef();

if (!normalizedViewTypeItems.current) {
  normalizedViewTypeItems.current = normalizeDropdownItems(viewTypeItems)
} 
``` 

Another option would be to use a useState with a function, but I think it is not a good fit either:

```javascript
const [normalizedViewTypeItems] = useState(() => normalizeDropdownItems(viewTypeItems))
```

Reply:
Oh, you are right, `normalize` in `useRef` does get called on each render. I always thought it doesn't, because it does not change current value, even if `viewTypeItems` change. `ref`s are weird.

About `useState`, I agree with you, it's not ideal. State variables have special meaning in components, it doesn't seem a good practice to hack different meaning into one.

`useMemo` and the above `useRef` seem equally good to me, just a matter of personal preference. If you like `useMemo` more, :+1: